server/src/main/java/io/crate/expression/predicate/IsNullPredicate.java

```diff
@@ -130,7 +130,9 @@ public static Query refExistsQuery(Reference ref, Context context, boolean count
                         .add(Queries.not(isNullFuncToQuery(ref, context)), Occur.SHOULD)
                         .build();
                 } else {
-                    return null;
+                    return new BooleanQuery.Builder()
+                        .add(Queries.not(isNullFuncToQuery(ref, context)), Occur.MUST)
+                        .build();
```

Comment:
`compare_run` `array_length` spec locally against `master` with the following local changes:
```
 statements = [
-  "create table t_arrays_biased_empty (xs array(integer))",
-  "create table t_arrays_biased_one_element (xs array(integer))",
+  "create table t_arrays_biased_empty (xs array(integer) storage with (columnstore=false))",
+  "create table t_arrays_biased_one_element (xs array(integer) storage with(columnstore=false))",
/t_arrays_biased_empty.json.gz' with (compression = 'gzip')",
@@ -11,63 +11,13 @@ statements = [
 ]
 
 [[queries]]
-statement = "select count(*) from t_arrays_biased_empty where xs = []"
+statement = "select count(*) from t_arrays_biased_empty where xs != []"
 iterations = 200
 
 [[queries]]
-statement = "select count(*) from t_arrays_biased_5_elements where xs = []"
+statement = "select count(*) from t_arrays_biased_5_elements where xs != []"
 iterations = 200
```

The outcome:
```
# Results (server side duration in ms)
V1: 5.8.0-1ab0c98318e1bb8ee1d5d96fd4e570cf2fea251e
V2: 5.8.0-129862e7da55941aa061cdf2225129cd26554506

Q: select count(*) from t_arrays_biased_empty where xs != []
C: 1
| Version |         Mean ±    Stdev |        Min |     Median |         Q3 |        Max |
|   V1    |       63.740 ±  343.430 |     24.687 |     26.643 |     28.614 |   4592.312 |
|   V2    |      320.778 ±  548.168 |    249.506 |    257.434 |    268.213 |   7512.844 |
├---------┴-------------------------┴------------┴------------┴------------┴------------┘
|               + 133.69%                           + 162.48%   
There is a 100.00% probability that the observed difference is not random, and the best estimate of that difference is 133.69%
The test has statistical significance

Q: select count(*) from t_arrays_biased_5_elements where xs != []
C: 1
| Version |         Mean ±    Stdev |        Min |     Median |         Q3 |        Max |
|   V1    |       29.974 ±  273.128 |      6.042 |      9.883 |     10.100 |   3866.242 |
|   V2    |       97.938 ±  410.621 |     44.845 |     47.135 |     48.827 |   4269.922 |
├---------┴-------------------------┴------------┴------------┴------------┴------------┘
|               + 106.27%                           + 130.66%   
There is a 94.80% probability that the observed difference is not random, and the best estimate of that difference is 106.27%
The test has no statistical significance
```

Similar to https://github.com/crate/crate/pull/16237 over 100% degradations.

Reply:
There are also some gains in different scenarios.

Ran `array_length` spec locally against `master` with the following local changes:
```
-  "create table t_arrays_biased_empty (xs array(integer))",
-  "create table t_arrays_biased_one_element (xs array(integer))",
+  "create table t_arrays_biased_empty (xs array(integer) storage with (columnstore=false))",
+  "create table t_arrays_biased_one_element (xs array(integer) storage with(columnstore=false))",
/t_arrays_biased_empty.json.gz' with (compression = 'gzip')",
@@ -11,63 +11,13 @@ statements = [
 ]
 
 [[queries]]
-statement = "select count(*) from t_arrays_biased_empty where xs = []"
+statement = "select count(*) from t_arrays_biased_empty where xs is not null"
 iterations = 200
 
 [[queries]]
-statement = "select count(*) from t_arrays_biased_5_elements where xs = []"
+statement = "select count(*) from t_arrays_biased_5_elements where xs is not null"
```

The outcome:
```
# Results (server side duration in ms)
V1: 5.8.0-1ab0c98318e1bb8ee1d5d96fd4e570cf2fea251e
V2: 5.8.0-129862e7da55941aa061cdf2225129cd26554506

Q: select count(*) from t_arrays_biased_empty where xs is not null
C: 1
| Version |         Mean ±    Stdev |        Min |     Median |         Q3 |        Max |
|   V1    |      247.654 ±  428.675 |    187.848 |    196.713 |    203.934 |   5963.568 |
|   V2    |      185.129 ±  493.634 |    129.237 |    133.938 |    143.269 |   6978.358 |
├---------┴-------------------------┴------------┴------------┴------------┴------------┘
|               -  28.89%                           -  37.97%   
There is a 82.30% probability that the observed difference is not random, and the best estimate of that difference is 28.89%
The test has no statistical significance

Q: select count(*) from t_arrays_biased_5_elements where xs is not null
C: 1
| Version |         Mean ±    Stdev |        Min |     Median |         Q3 |        Max |
|   V1    |      478.471 ±  443.101 |    398.338 |    411.335 |    423.153 |   3631.423 |
|   V2    |      500.095 ±  447.030 |    419.927 |    430.938 |    437.760 |   3674.234 |
├---------┴-------------------------┴------------┴------------┴------------┴------------┘
|               +   4.42%                           +   4.65%   
There is a 37.27% probability that the observed difference is not random, and the best estimate of that difference is 4.42%
The test has no statistical significance
```